packages/components/src/DataTable/Item/DataTableItem.tsx

```diff
@@ -77,26 +78,25 @@ const DataTableItemLayout: FC<DataTableItemProps> = ({
     ? handleOnSelect
     : onClick || undefined
 
-  const handleMenuClick = (event: React.MouseEvent<HTMLElement>) => {
-    event.stopPropagation()
-  }
-
   const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
     const isEventFromChild = event.currentTarget !== event.target
     if (event.keyCode === 13 && !isEventFromChild) {
       event.currentTarget.click()
     }
   }
 
-  const itemActions = actions && (
-    <ItemTarget onClick={handleMenuClick}>
-      <Menu>
-        <MenuDisclosure tooltip={actionsTooltip}>
-          <IconButton icon="DotsVert" size="small" label={actionsTooltip} />
-        </MenuDisclosure>
-        <MenuList>{actions}</MenuList>
-      </Menu>
-    </ItemTarget>
+  const itemActions = (
+    <ItemActionsLayout>
+      {primaryAction && <>{primaryAction}</>}
+      {actions && (
+        <Menu>
+          <MenuDisclosure tooltip={actionsTooltip}>
+            <IconButton icon="DotsVert" size="small" label={actionsTooltip} />
+          </MenuDisclosure>
+          <MenuList>{actions}</MenuList>
+        </Menu>
+      )}
+    </ItemActionsLayout>
```

Comment:
I'd prefer to the layout behavior built into the `DataTableRow`

```suggestion
  const itemActions = primaryAction || actions && (
    <>
      {primaryAction}
      {actions && (
        <Menu>
          <MenuDisclosure tooltip={actionsTooltip}>
            <IconButton icon="DotsVert" size="small" label={actionsTooltip} />
          </MenuDisclosure>
          <MenuList>{actions}</MenuList>
        </Menu>
      )}
    </>
  )
```